pkg/configurator/client_test.go

```diff
@@ -120,3 +123,151 @@ var _ = Describe("Test OSM ConfigMap parsing", func() {
+
+	confChannel := events.GetPubSubInstance().Subscribe(
+		announcements.ConfigMapAdded,
+		announcements.ConfigMapDeleted,
+		announcements.ConfigMapUpdated)
+	defer events.GetPubSubInstance().Unsub(confChannel)
+
+	configMap := v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: osmNamespace,
+			Name:      osmConfigMapName,
+		},
+		Data: make(map[string]string),
+		// All false
+	}
+	if _, err := kubeClient.CoreV1().ConfigMaps(osmNamespace).Create(context.TODO(), &configMap, metav1.CreateOptions{}); err != nil {
+		GinkgoT().Fatalf("[TEST] Error creating ConfigMap %s/%s/: %s", configMap.Namespace, configMap.Name, err.Error())
+	}
+	<-confChannel
+
+	proxyBroadcastChannel := events.GetPubSubInstance().Subscribe(announcements.ScheduleProxyBroadcast)
+	defer events.GetPubSubInstance().Unsub(proxyBroadcastChannel)
+
+	tests := []struct {
+		deltaConfigMapContents map[string]string
+		expectProxyBroadcast   bool
+	}{
+		{
+			deltaConfigMapContents: map[string]string{
+				egressKey: "true",
+			},
+			expectProxyBroadcast: true,
+		},
+		{
+			deltaConfigMapContents: map[string]string{
+				PermissiveTrafficPolicyModeKey: "true",
+			},
+			expectProxyBroadcast: true,
+		},
+		{
+			deltaConfigMapContents: map[string]string{
+				useHTTPSIngressKey: "true",
+			},
+			expectProxyBroadcast: true,
+		},
+		{
+			deltaConfigMapContents: map[string]string{
+				tracingEnableKey: "true",
+			},
+			expectProxyBroadcast: true,
+		},
+		{
+			deltaConfigMapContents: map[string]string{
+				tracingAddressKey: "true",
+			},
+			expectProxyBroadcast: true,
+		},
+		{
+			deltaConfigMapContents: map[string]string{
+				tracingEndpointKey: "true",
+			},
+			expectProxyBroadcast: true,
+		},
+		{
+			deltaConfigMapContents: map[string]string{
+				tracingPortKey: "true",
+			},
+			expectProxyBroadcast: true,
+		},
+		{
+			deltaConfigMapContents: map[string]string{
+				prometheusScrapingKey: "true",
+			},
+			expectProxyBroadcast: true,
+		},
+		{
+			deltaConfigMapContents: map[string]string{
+				configResyncInterval: "24h",
+			},
+			expectProxyBroadcast: false,
+		},
+		{
+			deltaConfigMapContents: map[string]string{
+				envoyLogLevel: "warn",
+			},
+			expectProxyBroadcast: false,
+		},
+		{
+			deltaConfigMapContents: map[string]string{
+				enableDebugServer: "true",
+			},
+			expectProxyBroadcast: false,
+		},
+		{
+			deltaConfigMapContents: map[string]string{
+				serviceCertValidityDurationKey: "30h",
+			},
+			expectProxyBroadcast: false,
+		},
+		{
+			deltaConfigMapContents: map[string]string{
+				serviceCertValidityDurationKey: "30h",
+			},
+			expectProxyBroadcast: false,
+		},
+		{
+			deltaConfigMapContents: map[string]string{
+				enablePrivilegedInitContainer: "true",
+			},
+			expectProxyBroadcast: false,
+		},
+		{
+			deltaConfigMapContents: map[string]string{
+				outboundIPRangeExclusionListKey: "true",
+			},
+			expectProxyBroadcast: false,
+		},
+	}
+
+	for _, t := range tests {
```

Comment:
@eduser25 It'd be great to use `test` instead of `t` (as a general rule) for readability.